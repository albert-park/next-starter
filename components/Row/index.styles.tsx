import css from 'styled-jsx/css';

export default css`
  .row {
    max-width: 880px;
    margin: 80px auto 40px;
    display: flex;
    flex-direction: row;
    justify-content: space-around;
  }
`;
