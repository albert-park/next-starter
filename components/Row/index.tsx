import React from 'react';
import { StyledRow } from './index.styles';

const Row = ({ children }) => (
  <StyledRow>
    {children}
  </StyledRow>
);

export default Row;
