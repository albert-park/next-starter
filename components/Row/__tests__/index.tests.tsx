import React from 'react';
import { render, getByTestId } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Row from '../index';
import Link from 'next/link';

const renderTest = () => {
  return (
    render(
    <Row>
      <Link href="/">
        <a href="/" data-testid="test-child">Test Link</a>
      </Link>
    </Row>
    )
  )
}
  
describe('Layout Component', () => {
  it('should match snapshot', () => {
    const { asFragment } = renderTest();
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders all children', () => {
    const { container, getByTestId } = renderTest();
    expect(container.children.length).toBeGreaterThan(0);
    expect(getByTestId("test-child")).toBeDefined();
  })
});
