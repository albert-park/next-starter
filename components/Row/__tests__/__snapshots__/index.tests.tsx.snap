// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="indexstyles__StyledRow-sc-2nh6n1-0 dWwEEj"
  >
    <a
      data-testid="test-child"
      href="/"
    >
      Test Link
    </a>
  </div>
</DocumentFragment>
`;
